### ISSUE TITLE
tp.file.cursor should change the mode to edit mode if in reading mode

### ISSUE THREAD

**Issue by RonMcKay** [SilentVoid13/Templater#1232]
**Plugin information (please complete the following information):**

-   OS: MacOS Ventura (13.6)
-   Templater version: 1.16.4
-   Obsidian version: 1.4.16
-   Templater settings:
    Template folder location: "templates/"
    Automatic jump to cursor: on
    Trigger Templater on new file creation: on
-   Periodic Notes settings:
    Daily Notes: on
    Format: `YYYY-MM-DD-dddd`
    Daily Note Template: `templates/daily-template.md`
    Note Folder: `diary`

Daily note template:
```txt
---
created: <% tp.file.creation_date() %>
---

<% tp.file.cursor(1) %>
```

To reproduce:

- Go to an arbitrary page and switch to reading view
- Execute the "Periodic Notes: Open daily note" command

These steps worked for me in a new vault with the minimal setup from above.

**Describe the bug**
When I trigger the "Periodic Notes: Open daily note" command from the active tab which is in reading view, the diary page is getting created and commands are replaced, except for `<% tp.file.cursor(1) %>`. This command remains on the diary page without being replaced.

**Expected behavior**
I understand that there can not be a cursor position when the diary page is in reading view, but I expect the templater code to be replaced with nothing.

**Screenshots**
This is the result of the daily note creation. The creation date is replaced, but `<% tp.file.cursor(1) %>` remains as is.
![image](https://github.com/SilentVoid13/Templater/assets/36410367/d173d809-dcc9-44a0-a488-e5586c53ae41)

**Comment by RonMcKay** [comment]
May be related to #1112

**Comment by spenserpothier** [comment]
I am experiencing this as well with same versions but macOS 13.4.1 (a).
It also happens when I try to create a periodic note from editing view.

**Comment by Zachatoo** [comment]
I'd argue that this is expected and desired behavior. It would be weird if a user setup multiple cursors and suddenly they're all gone because they were accidentally in reading mode instead of editing mode.

> [!NOTE]
> Example of multiple cursors.
> ```
> Name: <% tp.file.cursor(0) %>
> Status: <% tp.file.cursor(1) %>
> ```

I recommend switching to edit mode and run the `Templater: Jump to next cursor location` command to jump to the cursor and remove it.

**Comment by RonMcKay** [comment]
True, for multiple cursors this would be weird. But even for a single cursor with `<% tp.file.cursor(0) %>` this happens for me.
But maybe it is just me finding this weird. 😄

**Comment by davidolrik** [comment]
I also have this issue.

When creating a new node from the calendar widget, and either of these conditions are pressent:

- No files are open
- Nne or more are open, but not in edit mode

Then `<% tp.file.cursor() %>` is not executed.

I propose that when creating a new note and Templater is engaged, then it should switch to edit mode automatically.

**Comment by davidolrik** [comment]
Maybe `<% tp.file.cursor() %>` should also do that, as it only makes sense to use in edit mode.

**Comment by Fred-Vatin** [comment]
In the meantime, one can add a user function following the [doc](https://silentvoid13.github.io/Templater/user-functions/script-user-functions.html).
![image](https://github.com/SilentVoid13/Templater/assets/22337329/c7af667a-2c7f-4c80-8cab-6c872f5d4e95)

Script above exporting the function:
```js
function switch_to_editmode () {
	/* Be sure to switch to edit mode first (live preview)
		 It is required so it can insert the template in existing file. */
	const view = app.workspace.activeLeaf.getViewState()
	view.state.mode = 'source'
	view.state.source = false
	app.workspace.activeLeaf.setViewState(view)
}
module.exports = switch_to_editmode;
```

Then you call it in your template:

```js
<%*
/* Be sure to switch to edit mode first (live preview)
   It is required so it can insert the template in existing file. */
tp.user.Switch_to_edit_mode();

// start of template construct
// insert your logic here

if (title.startsWith("Untitled")) { 
  title = await tp.system.prompt("What is the plugin name ?"); 
  await tp.file.rename(title);
} 

// and so on
%>
```

Further in the template, add this to jump to the cursor on any case. It is necessary if create a new note from the template where the focus is on the note title. You just press enter to validate it and the cursor jumps to the anchor.
```md
# Description

<% tp.file.cursor(1) %><%* app.workspace.activeLeaf.view.editor?.focus(); %>
```

Source: https://forum.obsidian.md/t/easily-switch-between-source-mode-live-preview-preview/27151/25
